### PR TITLE
MBS-9094: Implement reset userdata

### LINKED ISSUE #92: Implement reset_userdata callback
As mootimeter is saving user data, it needs to implement the reset_userdata callback, so the userdata can be properly cleared when the course is being reset.

### PR DESCRIPTION
**Pull request by PhMemmel**
Closes #92

### CODE REVIEW

**Comment by pmaneggia** [comment]
I think the solution of having a uniform handling of the reset in the tool helper class is very practical and I do not object. In principle though one would think it is up to each plugin to decide this sort of behaviour, particularly how to delete the answers? This can however of course be addressed in the future.

**Comment by PhMemmel** [comment]
I agree and this is already possible: I intentionally created the corresponding functions in the toolhelper class and did not do it directly in the lib.php hooks, because the tools inherit from toolhelper class and - if they have special requirements for implementing these functions - can simply overwrite these methods from the toolhelper base class.

File: classes/toolhelper.php [hunk 1]
```diff
@@ -141,4 +142,55 @@ public function is_option_selected(int $optionid, object $config, string $attrib
...
+     */
+    public function reset_course_form_defaults(stdClass $course): array {
+        $shortclassname = preg_replace('/.*\\\\/', '', get_class($this));
+        return [
+            'reset_mootimetertool_' . $shortclassname . '_answers' => 1,
+        ];
+    }
+
+    /**
+     *
```
[the remark this answers is not in the document]
Comment (PhMemmel) [reply classes/toolhelper.php:173]: I missed this somehow, thanks

File: lang/en/mootimeter.php [hunk 2]
```diff
@@ -66,11 +66,13 @@
 $string['please_select_a_page'] = 'Please select a page, to answer a question';
 $string['pluginadministration'] = 'Mootimeter administration';
 $string['pluginname'] = 'Mootimeter';
+$string['pluginnameplural'] = 'Mootimeter activities';
 $string['privacy:metadata:mootimetertoolpluginsummary'] = 'Mootimetertools subplugin summary';
 $string['privacy:pagepath'] = 'Path of page with id: {$a}';
 $string['question'] = 'Question';
 $string['refreshinterval'] = 'Refresh interval';
 $string['refreshinterval_desc'] = 'Refresh interval in ms for result polling';
+$string['resetuserdata'] = 'Nutzerdaten löschen für {$a}';
```
[the remark this answers is not in the document]
Comment (PhMemmel) [reply lang/en/mootimeter.php]: changed, thx

File: lib.php [hunk 3]
```diff
@@ -243,3 +243,54 @@ function mootimeter_inplace_editable($itemtype, $itemid, $newvalue) {
...
+/**
+ * Callback function for resetting the user data in a course.
+ *
+ * @param stdClass $data the data submitted by the related mform
+ * @return array array of associative arrays representing the status of the performed actions
+ */
+function mootimeter_reset_userdata(stdClass $data): array {
+    $status = [];
+
+    foreach (\mod_mootimeter\plugininfo\mootimetertool::get_enabled_plugins() as $tool) {
```
[the remark this answers is not in the document]
Comment (PhMemmel) [reply lib.php]: good catch, changed it

**Review by PhMemmel — COMMENTED** [review]
Adressed all comments and added a commit which should be squashed when merging

**Pull request merged**